saleor/product/utils.py

```diff
@@ -377,3 +351,56 @@ def get_product_list_context(request, filter_set):
         'sort_by_choices': SORT_BY_FIELDS,
         'now_sorted_by': now_sorted_by,
         'is_descending': is_descending}
+
+
+def get_product_attributes_data(product):
```

Comment:
This `utils` module has become quite cluttered, so maybe we should think of splitting it into multiple files. E.g., these attributes-related functions could be stored in a separate file `attributes.py`. I'm thinking of such structure:
```
product
  utils
    __init__.py
    attributes.py
    availability.py
    variant_picker.py
    etc.
```
For now, we could only introduce `attributes.py` and keep the rest of utility functions in `__init__.py` (which could import utilities from all submodules) and clean up the rest in a separate PR. Up to you.